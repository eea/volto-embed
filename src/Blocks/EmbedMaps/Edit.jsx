import React from 'react';
import { compose } from 'redux';
import { injectIntl } from 'react-intl';
import BlockDataForm from '@plone/volto/components/manage/Form/BlockDataForm';
import { SidebarPortal } from '@plone/volto/components';
import View from './View';
import getSchema from './schema';

const Edit = (props) => {
  const { block, data, selected, onChangeBlock } = props;
  const schema = React.useMemo(() => getSchema(props), [props]);

  return (
    <React.Fragment>
      <View {...props} mode="edit" />
      <SidebarPortal selected={selected}>
        <BlockDataForm
          block={block}
          schema={schema}
          title={schema.title}
          onChangeField={(id, value) => {
            onChangeBlock(block, {
              ...data,
              [id]: value,
            });
          }}
          onChangeBlock={onChangeBlock}
          formData={data}
        />
      </SidebarPortal>
    </React.Fragment>
  );
};

export default compose(injectIntl)(Edit);
