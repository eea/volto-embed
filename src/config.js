import ViewIframeBlock from './Iframe/ViewIframe.jsx';
import EditIframeBlock from './Iframe/EditIframe.jsx';

export function applyConfig(config) {
  config.blocks.blocksConfig.maps = {
    ...config.blocks.blocksConfig.maps,
    view: ViewIframeBlock,
    edit: EditIframeBlock,
  };
  return config;
}
