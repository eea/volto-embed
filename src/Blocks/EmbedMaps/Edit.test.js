import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Provider } from 'react-intl-redux';
import config from '@plone/volto/registry';

import Edit from './Edit';
import installEmbedMaps from '.';
import '@testing-library/jest-dom/extend-expect';

installEmbedMaps(config);

jest.mock('@plone/volto/components', () => ({
  SidebarPortal: ({ children }) => (
    <div data-testid="sidebar-portal">{children}</div>
  ),
  UniversalLink: ({ children, href }) => <a href={href}>{children}</a>,
}));

jest.mock(
  '@plone/volto/components/manage/Form/BlockDataForm',
  () => (props) => (
    <div data-testid="block-data-form">
      <p>Mocked BlockDataForm</p>
      <input data-testid="block-data-input" onChange={props.onChangeField} />
    </div>
  ),
);

jest.mock('./View', () =>
  jest.fn((props) => (
    <div>
      <div data-testid="view-component" mode={props.mode}>
        Mocked View Component
      </div>
      {props.children}
    </div>
  )),
);

describe('Edit component', () => {
  const mockOnChangeBlock = jest.fn();
  const props = {
    id: 'my-map',
    data: {
      '@type': 'embed_maps',
      with_notes: false,
      with_sources: false,
      with_more_info: true,
      with_share: true,
      with_enlarge: true,
      url: '/path/to/map',
      maps: {
        '@id': '/path/to/map',
        title: 'My map',
        url: 'https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d3027.7835278268726!2d14.38842915203974!3d40.634655679238854!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x133b994881d943cb%3A0x6ab93db57d3272f0!2sHotel+Mediterraneo+Sorrento!5e0!3m2!1sen!2ses!4v1550168740166',
      },
      dataprotection: {
        enabled: false,
      },
      useVisibilitySensor: false,
    },
    pathname: '/news',
    selected: true,
    block: '1234',
    index: 1,
    onChangeBlock: jest.fn(),
    onSelectBlock: jest.fn(),
    onDeleteBlock: jest.fn(),
    onFocusPreviousBlock: jest.fn(),
    onFocusNextBlock: jest.fn(),
    handleKeyDown: jest.fn(),
    content: {},
  };

  it('should render View component with mode set to edit', () => {
    const { getByTestId } = render(
      <Provider store={global.store}>
        <Edit {...props} />
      </Provider>,
    );

    const viewComponent = getByTestId('view-component');
    expect(viewComponent).toBeInTheDocument();
    expect(viewComponent).toHaveAttribute('mode', 'edit');
  });

  it('should render SidebarPortal when selected is true', () => {
    const { getByTestId } = render(
      <Provider store={global.store}>
        <Edit {...props} />
      </Provider>,
    );
    const sidebarPortal = getByTestId('sidebar-portal');
    expect(sidebarPortal).toBeInTheDocument();
  });

  it('should render SidebarPortal and BlockDataForm', () => {
    const { getByTestId } = render(
      <Provider store={global.store}>
        <Edit {...props} />
      </Provider>,
    );

    const sidebarPortal = getByTestId('sidebar-portal');
    const blockDataForm = getByTestId('block-data-form');
    expect(sidebarPortal).toBeInTheDocument();
    expect(blockDataForm).toBeInTheDocument();
  });

  it('should call onChangeBlock when onChangeField is triggered', () => {
    const { getByTestId } = render(
      <Provider store={global.store}>
        <Edit {...props} onChangeBlock={mockOnChangeBlock} />
      </Provider>,
    );

    const inputField = getByTestId('block-data-input');
    fireEvent.change(inputField, { target: { value: 'new value' } });

    expect(mockOnChangeBlock).toHaveBeenCalled();
  });
});
