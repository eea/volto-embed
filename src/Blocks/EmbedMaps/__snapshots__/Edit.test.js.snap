// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an edit embed map block component 1`] = `
Array [
  <div
    className="embed-map"
  >
    <div
      className="block maps align center"
    >
      <div
        className="maps-inner"
      >
        <div
          className="privacy-protection-wrapper"
          style={
            Object {
              "height": "auto",
              "minHeight": "200px",
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <iframe
            allowFullScreen={true}
            className="google-map"
            frameBorder="0"
            src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d3027.7835278268726!2d14.38842915203974!3d40.634655679238854!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x133b994881d943cb%3A0x6ab93db57d3272f0!2sHotel+Mediterraneo+Sorrento!5e0!3m2!1sen!2ses!4v1550168740166"
            style={Object {}}
            title="Embeded Google Maps"
          />
        </div>
      </div>
      <div
        className="visualization-toolbar"
      >
        <div
          className="left-col"
        >
          <div
            className="more-info"
          >
            <a
              href="/path/to/map"
            >
              <span>
                More info
              </span>
              <i
                class="ri-external-link-line"
              />
            </a>
          </div>
        </div>
        <div
          className="right-col"
        >
          <div
            className="popup"
          >
            <div
              className="trigger"
            >
              <div
                className="share"
              >
                <button
                  className="trigger-button"
                >
                  <i
                    class="ri-share-fill"
                  />
                  <span>
                    Share
                  </span>
                </button>
              </div>
            </div>
            <div
              className="content"
            >
              <div
                className="item"
              >
                <span
                  className="label"
                >
                  Copy link
                </span>
                <div
                  className="control"
                >
                  <div
                    className="ui input share-link"
                  >
                    <input
                      onChange={[Function]}
                      type="text"
                      value="/path/to/map"
                    />
                  </div>
                  <button
                    className="ui primary button copy-button copy-button"
                    onClick={[Function]}
                  >
                    Copy
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            className="enlarge"
          >
            <button
              className="trigger-button"
              onClick={[Function]}
            >
              <i
                className="ri-fullscreen-line"
              />
              Enlarge
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    id="sidebar"
  >
    <div
      className="ui form"
    >
      <header
        className="header pulled"
      >
        <h2>
          Embed interactive Map
        </h2>
      </header>
      <div
        id="blockform-fieldset-default"
      >
        <div
          className="ui segment form attached"
        >
          <div
            className="mocked-url-widget"
            id="mocked-field-url"
          >
            Map url
             - 
            No description
          </div>
          <div
            className="mocked-default-widget"
            id="mocked-field-height"
          >
            <a
              href="https://developer.mozilla.org/en-US/docs/Web/CSS/height"
              rel="noreferrer"
              target="_blank"
            >
              CSS height
            </a>
             - 
            Map height
          </div>
        </div>
      </div>
      <div
        className="accordion ui fluid styled form"
      >
        <div
          id="blockform-fieldset-toolbar"
        >
          <div
            className="active title"
            onClick={[Function]}
          >
            Toolbar
            <svg
              className="icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": undefined,
                }
              }
              onClick={null}
              style={
                Object {
                  "fill": "currentColor",
                  "height": "20px",
                  "width": "auto",
                }
              }
              viewBox=""
              xmlns=""
            />
          </div>
          <div
            className="content active"
          >
            <div
              aria-hidden={false}
              className="rah-static rah-static--height-auto"
              style={
                Object {
                  "height": "auto",
                  "overflow": "visible",
                }
              }
            >
              <div
                style={
                  Object {
                    "WebkitTransition": "opacity 500ms ease 0ms",
                    "transition": "opacity 500ms ease 0ms",
                  }
                }
              >
                <div
                  className="ui segment attached"
                >
                  <div
                    className="mocked-boolean-widget"
                    id="mocked-field-with_notes"
                  >
                    Show note
                     - 
                    No description
                  </div>
                  <div
                    className="mocked-boolean-widget"
                    id="mocked-field-with_sources"
                  >
                    Show sources
                     - 
                    Will show sources set in this page Data provenance
                  </div>
                  <div
                    className="mocked-boolean-widget"
                    id="mocked-field-with_more_info"
                  >
                    Show more info
                     - 
                    No description
                  </div>
                  <div
                    className="mocked-boolean-widget"
                    id="mocked-field-with_share"
                  >
                    Show share button
                     - 
                    No description
                  </div>
                  <div
                    className="mocked-boolean-widget"
                    id="mocked-field-with_enlarge"
                  >
                    Show enlarge button
                     - 
                    No description
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
